src/holofuel/pages/Inbox/Inbox.js

```diff
@@ -485,9 +483,12 @@ export function DeclinedTransactionModal ({ handleClose, isDeclinedTransactionMo
   </Modal>
 }
 
-export function ConfirmationModal ({ transaction, handleClose, declineTransaction, refundTransaction, payTransaction, acceptOffer, setCounterpartyNotFound, counterpartyNotFound, setHasTransactionBeenActioned }) {
+export function ConfirmationModal ({ confirmationModalProperties, setNewModalTransactionProperties, clearHighlightedTransaction, declineTransaction, refundTransaction, payTransaction, acceptOffer, setCounterpartyNotFound, counterpartyNotFound }) {
   const { newMessage } = useFlashMessageContext()
-  const { id, amount, type, action, notes } = transaction
+  const { transactions, action, shouldDisplay } = confirmationModalProperties
+  const transaction = transactions[0] || {}
```

Comment:
why are we passing in a list of transactions here if we're only ever using the first one?

Reply:
We're passing a list in the `DeclinedTransactionModal` so we can resuse the same function to update the actined-transaction in the row, as the only difference *is* that this one contains a list of tx.

Once we rid of this modal, we can simplify the functionality.